File: Array/Number of Digit One.go
package main

import "fmt"

func main() {

	A := 100
	result := countOne(A)
	fmt.Println(result)
}

func countOne(A int) int {
	ans := 0

	for i := 1; i <= A; i *= 10 {
		ptr := i * 10
		ans += (A/ptr)*i + min(max(A%ptr-i+1, 0), i)
	}

	return ans
}

func max(a, b int) int {
	if a > b {
		return a
	}
	return b
}

func min(a, b int) int {
	if a < b {
		return a
	}
	return b
}

File: Pythons Solutions/Number of Digit One.py
class Solution:
    def digitOne(self, num):
        ans  = 0
        i = 1
        while i <= num:
            div = i * 10
            ans += (num//div)*i + min(max(num%div - i + 1, 0), i)
            i = i * 10

        return int(ans)

obj = Solution()
print(obj.digitOne(10))